consumer.go

```diff
@@ -215,6 +215,12 @@ func (c *Consumer) ScanShard(ctx context.Context, shardID string, fn ScanFunc) e
 
 			if isShardClosed(resp.NextShardIterator, shardIterator) {
 				c.logger.Log("[CONSUMER] shard closed:", shardID)
+				if shardClosedHandler, ok := c.store.(ShardClosedHandler); ok {
```

Comment:
Lemme know if you want to keep this, or implement as a callback. 

The callback route is interesting and I may extend to adding a callback for each part of the lifecycle; similar to the log points it could be great place to add telemetry or additional functionality

Reply:
I think I like the separate callback idea better now. I'll close this PR and make one for this event that I'm interested in.